functions.h

```diff
@@ -324,10 +315,10 @@ auto bessel_i_helper(const T& z) {
         y_tmp = 2. * n / z * y0 + y1;
         y1 = y0;
         y0 = y_tmp;
-        mu += 2. * (n % 2 == 0 ? 1 : -1) * (z.real() < 0 ? -1 : 1) * y1;
```

Comment:
It seems to me this is totally different with before, but why we are still reasonable before?

Reply:
When z is small, this coefficient did not play an important role, I guess.